test/cmd/debug.sh

```diff
@@ -88,6 +88,44 @@ run_kubectl_debug_pod_tests() {
   # Clean up
   kubectl delete pod target target-copy "${kube_flags[@]:?}"
 
+  ### Pod Troubleshooting by ephemeral containers with debugging profile
+  
+  # Pre-Condition: Pod "nginx" is created
+  kubectl run target "--image=${IMAGE_NGINX:?}" "${kube_flags[@]:?}"
+  kube::test::get_object_assert pod "{{range.items}}{{${id_field:?}}}:{{end}}" 'target:'
+  # Restricted profile just works in not restricted namespace
+  # Command: add a new debug container with restricted profile
+  kubectl debug target -it --image=gcr.io/distroless/static-debian11:debug-nonroot --attach=false -c debug-container --profile=restricted "${kube_flags[@]:?}"
```

Comment:
Could you please use simpler image(i.e. busybox). As long as we have restricted namespace, we can still reproduce the true restricted environment without using this image?

Reply:
Restrected profile requires that the container runs as non-root user(`runAsNonRoot=true`).
It's necessary to use the container image which's user is non-root, because there is no way to indicate user when we start EphemeralContainer now.
So I use `gcr.io/distroless/static-debian11:debug-nonroot` which runs as non-root user.

In fact, since the test case does not actually start the container, the test itself works fine even if we use a container image that starts as the root user, such as `busybox`.

Should I use `busybox`?

Reply:
I understand that this test is to check if the EphemeralContainer can be added successfully.
Therefore, I judge there is no problem with testing using `busybox`, and use it.

Reply:
But critical point is test should fail without your patch and success with your patch